### PR TITLE
collab (badge, comment, browser): reorder methods

### PR DESCRIPTION
**Pull request by linusha**

### CODE REVIEW

File: lively.collab/comments/badge.js [hunk 1]
```diff
@@ -42,6 +45,9 @@ export class Badge extends Morph {
     }
   }
 
+  /*
+  MORPH INTEGRATION
```
Comment (frcroth) [inline lively.collab/comments/badge.js]:
```suggestion
  INTERACTING WITH MORPH
```

**Review by frcroth — APPROVED** [review]

File: lively.collab/comments/commentBrowser.js [hunk 2]
```diff
@@ -73,16 +63,35 @@ export class CommentBrowser extends Window {
...
+    return !!CommentBrowser.instance.showsResolvedComments;
   }
 
   static toggleArchive () {
     CommentBrowser.instance.toggleArchive();
   }
 
-  // Construction and initialization
+  /*
+  COMMENT CREATION AND MAINTENANCE
```
Comment (T4rikA) [inline lively.collab/comments/commentBrowser.js]:
```suggestion
  COMMENT CREATION AND UPDATE
```

File: lively.collab/comments/commentBrowser.js [hunk 3]
```diff
@@ -209,18 +231,9 @@ export class CommentBrowser extends Window {
...
-        return;
-      }
-      for (const comment of morph.comments) {
-        await this.addCommentForMorph(comment, morph);
-      }
-    });
-  }
-
+  /*
+  COMMENT CREATION AND MAINTENANCE
```
Comment (T4rikA) [inline lively.collab/comments/commentBrowser.js]:
```suggestion
  COMMENT CREATION AND UPDATE
```

File: lively.collab/comments/commentMorph.js [hunk 4]
```diff
@@ -30,6 +30,9 @@ export class CommentGroupMorph extends Morph {
     };
   }
 
+  /*
+  INIT
```
Comment (T4rikA) [inline lively.collab/comments/commentMorph.js]:
```suggestion
  INITIALIZATION
```

File: lively.collab/comments/commentMorph.js [hunk 5]
```diff
@@ -143,6 +160,9 @@ export class CommentMorph extends Morph {
...
```
Comment (T4rikA) [inline lively.collab/comments/commentMorph.js]:
```suggestion
  INITIALIZATION
```

File: lively.collab/comments/commentIndicator.js [hunk 6]
```diff
@@ -111,6 +102,27 @@ export class CommentIndicator extends Label {
...
+  */
+  abandon () {
+    const referenceMorph = this.referenceMorph;
+    disconnect(referenceMorph, 'onChange', this, 'referenceMoving');
+    disconnect(referenceMorph, 'onOwnerChanged', this, 'connectMorphs');
+    super.abandon();
+  }
+
+  get isCommentIndicator () {
+    return true;
```
Comment (frcroth) [inline lively.collab/comments/commentIndicator.js]: why is this at deletion copying? I know it is used in this context, but I don't think it fits here

File: lively.collab/comments/commentBrowser.js [hunk 7]
```diff
@@ -7,42 +7,20 @@ import { CommentMorph, Badge } from 'lively.collab';
...
-  static close () {
-    if (CommentBrowser.isOpen()) {
-      CommentBrowser.instance.close();
-    }
-  }
+  /*
+  //////
+  Static Methods
+  //////
+  */
```
Comment (SilvanVerhoeven) [inline lively.collab/comments/commentBrowser.js:14]:
```suggestion
  // -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
  // static methods
  // -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
```

File: lively.collab/comments/commentBrowser.js [hunk 8]
```diff
@@ -7,42 +7,20 @@ import { CommentMorph, Badge } from 'lively.collab';
...
 
+  // static methods are the external interface
```
Comment (SilvanVerhoeven) [inline lively.collab/comments/commentBrowser.js]:
```suggestion
```

File: lively.collab/comments/commentBrowser.js (hunk 7 above)
Comment (SilvanVerhoeven) [inline lively.collab/comments/commentBrowser.js]:
```suggestion
  /*
  //////
  Static Methods (external interface)
  //////
  */
```

File: lively.collab/comments/commentBrowser.js [hunk 9]
```diff
@@ -73,16 +63,35 @@ export class CommentBrowser extends Window {
     CommentBrowser.isOpen() ? CommentBrowser.close() : CommentBrowser.open();
   }
 
-  static async whenRendered () {
-    return CommentBrowser.instance.whenRendered();
+  // -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+  // browser mode
```
Comment (SilvanVerhoeven) [inline lively.collab/comments/commentBrowser.js]:
```suggestion
  // browser mode / archive
```

File: lively.collab/comments/commentBrowser.js [hunk 10]
```diff
@@ -119,6 +132,9 @@ export class CommentBrowser extends Window {
     return CommentBrowser.instance;
   }
 
+  // -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+  // ui init
```
Comment (SilvanVerhoeven) [inline lively.collab/comments/commentBrowser.js]:
```suggestion
  // ui initialization
```

File: lively.collab/comments/commentBrowser.js [hunk 11]
```diff
@@ -181,6 +197,9 @@ export class CommentBrowser extends Window {
     this.relayoutWindowControls();
   }
 
+  // -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+  // browser mode
```
Comment (SilvanVerhoeven) [inline lively.collab/comments/commentBrowser.js]:
```suggestion
  // browser mode / archive
```

File: lively.collab/comments/commentBrowser.js [hunk 12]
```diff
@@ -259,9 +272,19 @@ export class CommentBrowser extends Window {
     this.commentGroups[morph.id] && this.commentGroups[morph.id].updateName();
   }
 
-  async applyResolveStatus (comment, referenceMorph) {
-    await this.removeCommentForMorph(comment, referenceMorph);
-    await this.addCommentForMorph(comment, referenceMorph);
+  // -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+  // ui changes for comments
```
Comment (SilvanVerhoeven) [inline lively.collab/comments/commentBrowser.js]:
```suggestion
  // comment group creation/deletion
```

File: lively.collab/comments/commentBrowser.js [hunk 13]
```diff
@@ -270,6 +293,14 @@ export class CommentBrowser extends Window {
     group.remove();
   }
 
+  async applyResolveStatus (comment, referenceMorph) {
+    await this.removeCommentForMorph(comment, referenceMorph);
+    await this.addCommentForMorph(comment, referenceMorph);
+  }
```
Comment (SilvanVerhoeven) [inline lively.collab/comments/commentBrowser.js]: Maybe we could either put it into the `browser mode / archive` section or make a section `resolve comments`?

**Review by SilvanVerhoeven — CHANGES_REQUESTED** [review]

File: lively.collab/comments/commentIndicator.js (hunk 6 above)
Comment (SilvanVerhoeven) [reply lively.collab/comments/commentIndicator.js]: +1
Would put it just below the constructor, without any specific category

File: lively.collab/comments/commentIndicator.js [hunk 14]
```diff
@@ -64,6 +63,15 @@ export class CommentIndicator extends Label {
     }
   }
 
+  display () {
```
Comment (SilvanVerhoeven) [inline lively.collab/comments/commentIndicator.js:73]: maybe we make a `visibility` section for this and hide? We could put `abandon` into it as well or move `abandon` into a category `creation/deletion` that has `connectMorphs` and `canBeCopied` in it

**Review by SilvanVerhoeven — CHANGES_REQUESTED** [review]

**Review by SilvanVerhoeven — APPROVED** [review]

**Pull request merged**